### PR TITLE
Write KV health status for all services

### PR DESCRIPTION
**Pull request by bwester**
In this commit, even services that don't declare any health check will still
have a health check written to the KV with "passing" status. This happens to be
the same behavior that the catalog-based health checks have.

### CODE REVIEW

File: pkg/watch/health.go [hunk 1]
```diff
@@ -212,7 +214,25 @@ func (p *PodWatch) checkHealth() {
...
+		return sc.resultFromCheck(sc.StatusCheck())
+	} else {
+		// "unknown" is probably more accurate, but automated tools can't handle an app that is
+		// always non-"passing". For instance, p2-replicate by default waits for a node to
+		// become "passing" before it considers the deployment a success.
+		//
+		// TODO: P2 has the capacity to check whether the app's process is running. This would
+		// make a great default status check! However, that information isn't easily accessible
+		// over here in the watch package. It would take a lot of refactoring to make this
+		// happen.
```
Comment (anthonybishopric) [inline pkg/watch/health.go]: That would definitely be a good default check.

Comment (anthonybishopric) [inline pkg/watch/health.go]: Except that frequently people deploy apps intended to not run at all, even if they have running processes - basically using P2 as a tar delivery mechanism. Perhaps the right solution is to provide a generic runit monitor sidekick that can be added to a pod manifest

File: pkg/watch/health.go [hunk 2]
```diff
@@ -156,13 +156,15 @@ func updatePods(
...
-		if missing && man.Manifest.StatusPort != 0 {
+		if missing {
 			sc := StatusChecker{
 				ID:     man.Manifest.Id,
 				Node:   node,
 				Client: client,
 			}
-			if man.Manifest.StatusHTTP {
+			if man.Manifest.StatusPort == 0 {
+				sc.URI = ""
```
Comment (anthonybishopric) [inline pkg/watch/health.go]: Probably A Bad Idea ®: make the default URI an endpoint on the _preparer_ that effectively emits the result of `sv stat` for a given pod ID.

**Comment by anthonybishopric** [comment]
:ship:

File: pkg/watch/health.go (hunk 2 above)
Comment (bwester) [inline pkg/watch/health.go]: I don't even follow what you're suggesting.  But since it's a bad idea, we probably don't want to do it anyway.

Comment (anthonybishopric) [inline pkg/watch/health.go]: :+1:

**Pull request merged**